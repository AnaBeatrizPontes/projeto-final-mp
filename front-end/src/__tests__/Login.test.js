import React from 'react'
import { fireEvent, render, getByText } from '@testing-library/react'
import '@testing-library/jest-dom/extend-expect';
import renderer from 'react-test-renderer'

import PagesLogin from '../pages/Login/Login';


describe('Teste da pagina de login', () => {

    test('Apertar num botao sem preencher os inputs', async () => {
        const ComponenteRenderizado = renderer.create(<PagesLogin />).toJSON()
        expect(ComponenteRenderizado).toMatchSnapshot()
    });

    test('Show alert if btn is pressed', async () => {
        const ComponenteRenderizado = render(<PagesLogin />);
        const { getByText } = ComponenteRenderizado;
        const btn = getByText("Entrar");
        fireEvent(
            btn,
            new MouseEvent('click', {
                bubbles: true,
                cancelable: true,
            })
        );
    });

});
