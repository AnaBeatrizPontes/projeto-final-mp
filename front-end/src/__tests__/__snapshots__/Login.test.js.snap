// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Teste da pagina de login Apertar num botao sem preencher os inputs 1`] = `
<div
  className="back"
>
  <div
    className="login"
  >
    <div
      className="login-logo"
    >
      <img
        alt="Logo"
        src="logo.png"
      />
    </div>
    <div
      className="login-right"
    >
      <h1>
        Acessar App
      </h1>
      <div
        className="loginInputEmail"
      >
        <svg
          aria-hidden={true}
          className="MuiSvgIcon-root"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="M20 4H4c-1.1 0-1.99.9-1.99 2L2 18c0 1.1.9 2 2 2h16c1.1 0 2-.9 2-2V6c0-1.1-.9-2-2-2zm0 4l-8 5-8-5V6l8 5 8-5v2z"
          />
        </svg>
        <input
          onChange={[Function]}
          placeholder="Digite um email"
          type="email"
          value=""
        />
      </div>
      <div
        className="loginInputPassword"
      >
        <svg
          aria-hidden={true}
          className="MuiSvgIcon-root"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="M18 8h-1V6c0-2.76-2.24-5-5-5S7 3.24 7 6v2H6c-1.1 0-2 .9-2 2v10c0 1.1.9 2 2 2h12c1.1 0 2-.9 2-2V10c0-1.1-.9-2-2-2zm-6 9c-1.1 0-2-.9-2-2s.9-2 2-2 2 .9 2 2-.9 2-2 2zm3.1-9H8.9V6c0-1.71 1.39-3.1 3.1-3.1 1.71 0 3.1 1.39 3.1 3.1v2z"
          />
        </svg>
        <input
          onChange={[Function]}
          placeholder="Digite sua senha"
          type="password"
          value=""
        />
        <div
          className="login-eye"
        >
          <svg
            aria-hidden={true}
            className="MuiSvgIcon-root"
            focusable="false"
            onClick={[Function]}
            viewBox="0 0 24 24"
          >
            <path
              d="M12 7c2.76 0 5 2.24 5 5 0 .65-.13 1.26-.36 1.83l2.92 2.92c1.51-1.26 2.7-2.89 3.43-4.75-1.73-4.39-6-7.5-11-7.5-1.4 0-2.74.25-3.98.7l2.16 2.16C10.74 7.13 11.35 7 12 7zM2 4.27l2.28 2.28.46.46C3.08 8.3 1.78 10.02 1 12c1.73 4.39 6 7.5 11 7.5 1.55 0 3.03-.3 4.38-.84l.42.42L19.73 22 21 20.73 3.27 3 2 4.27zM7.53 9.8l1.55 1.55c-.05.21-.08.43-.08.65 0 1.66 1.34 3 3 3 .22 0 .44-.03.65-.08l1.55 1.55c-.67.33-1.41.53-2.2.53-2.76 0-5-2.24-5-5 0-.79.2-1.53.53-2.2zm4.31-.78l3.15 3.15.02-.16c0-1.66-1.34-3-3-3l-.17.01z"
            />
          </svg>
        </div>
      </div>
      <div>
        <button
          className="mailButton"
          onClick={[Function]}
        >
          Entrar
        </button>
      </div>
      <div>
        <h4>
          Não tenho conta
        </h4>
      </div>
      <div>
        <button
          className="passButton"
          onClick={[Function]}
        >
          Cadastrar
        </button>
      </div>
    </div>
  </div>
</div>
`;

exports[`Teste da pagina de login v2 Show alert if btn is pressed with blank input v2 1`] = `
<div
  className="back"
>
  <div
    className="login"
  >
    <div
      className="login-logo"
    >
      <img
        alt="Logo"
        src="logo.png"
      />
    </div>
    <div
      className="login-right"
    >
      <h1>
        Acessar App
      </h1>
      <div
        className="loginInputEmail"
      >
        <svg
          aria-hidden={true}
          className="MuiSvgIcon-root"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="M20 4H4c-1.1 0-1.99.9-1.99 2L2 18c0 1.1.9 2 2 2h16c1.1 0 2-.9 2-2V6c0-1.1-.9-2-2-2zm0 4l-8 5-8-5V6l8 5 8-5v2z"
          />
        </svg>
        <input
          onChange={[Function]}
          placeholder="Digite um email"
          type="email"
          value=""
        />
      </div>
      <div
        className="loginInputPassword"
      >
        <svg
          aria-hidden={true}
          className="MuiSvgIcon-root"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="M18 8h-1V6c0-2.76-2.24-5-5-5S7 3.24 7 6v2H6c-1.1 0-2 .9-2 2v10c0 1.1.9 2 2 2h12c1.1 0 2-.9 2-2V10c0-1.1-.9-2-2-2zm-6 9c-1.1 0-2-.9-2-2s.9-2 2-2 2 .9 2 2-.9 2-2 2zm3.1-9H8.9V6c0-1.71 1.39-3.1 3.1-3.1 1.71 0 3.1 1.39 3.1 3.1v2z"
          />
        </svg>
        <input
          onChange={[Function]}
          placeholder="Digite sua senha"
          type="password"
          value=""
        />
        <div
          className="login-eye"
        >
          <svg
            aria-hidden={true}
            className="MuiSvgIcon-root"
            focusable="false"
            onClick={[Function]}
            viewBox="0 0 24 24"
          >
            <path
              d="M12 7c2.76 0 5 2.24 5 5 0 .65-.13 1.26-.36 1.83l2.92 2.92c1.51-1.26 2.7-2.89 3.43-4.75-1.73-4.39-6-7.5-11-7.5-1.4 0-2.74.25-3.98.7l2.16 2.16C10.74 7.13 11.35 7 12 7zM2 4.27l2.28 2.28.46.46C3.08 8.3 1.78 10.02 1 12c1.73 4.39 6 7.5 11 7.5 1.55 0 3.03-.3 4.38-.84l.42.42L19.73 22 21 20.73 3.27 3 2 4.27zM7.53 9.8l1.55 1.55c-.05.21-.08.43-.08.65 0 1.66 1.34 3 3 3 .22 0 .44-.03.65-.08l1.55 1.55c-.67.33-1.41.53-2.2.53-2.76 0-5-2.24-5-5 0-.79.2-1.53.53-2.2zm4.31-.78l3.15 3.15.02-.16c0-1.66-1.34-3-3-3l-.17.01z"
            />
          </svg>
        </div>
      </div>
      <div>
        <button
          className="mailButton"
          onClick={[Function]}
        >
          Entrar
        </button>
      </div>
      <div>
        <h4>
          Não tenho conta
        </h4>
      </div>
      <div>
        <button
          className="passButton"
          onClick={[Function]}
        >
          Cadastrar
        </button>
      </div>
    </div>
  </div>
</div>
`;
